Polyglot/AppDelegate.swift

```diff
@@ -0,0 +1,119 @@
+import Cocoa
+import KeyHolder
+import Magnet
+
+struct SettingsKey {
+  static let KeyCode = "keyCode"
+  static let KeyCodeUnicode = "keyCodeUnicode"
+  static let Modifiers = "modifiers"
+  static let SourceLanguage = "sourceLanguage"
+  static let TargetLanguage = "targetLanguage"
+  static let InstantTranslation = "instantTranslation"
+}
+
+let DEFAULT_SOURCE = ("auto", "Auto Detect")
+let DEFAULT_TARGET = ("en", "English")
+let GROUP_ID = "58XDWHK3JX.io.uechi.Polyglot"
+
+@NSApplicationMain
+class AppDelegate: NSObject, NSApplicationDelegate {
+  @IBOutlet var window: NSWindow!
+  @IBOutlet var recordView: RecordView! {
+    didSet {
+      if #available(OSX 10.14, *) {
+        recordView.tintColor = .controlAccentColor
+        recordView.backgroundColor = .controlBackgroundColor
+        recordView.borderColor = .controlShadowColor
+      }
+    }
+  }
+
+  @IBOutlet var sourceLanguagePopup: NSPopUpButton!
+  @IBOutlet var targetLanguagePopup: NSPopUpButton!
+  @IBOutlet var instantTranslation: NSButton!
+
+  var settings = UserDefaults(suiteName: GROUP_ID)!
+
+  func applicationDidFinishLaunching(_: Notification) {
+    setupPopupButtons()
+    setupKeyComboView()
+    setupInstantCheckbox()
+  }
+
+  func setupPopupButtons() {
+    let sourceLanguages = Constants.getSourceLabels()
+    let targetLanguages = Constants.getTargetLabels()
+    sourceLanguagePopup.addItems(withTitles: sourceLanguages)
+    targetLanguagePopup.addItems(withTitles: targetLanguages)
+
+    sourceLanguagePopup.target = self
+    targetLanguagePopup.target = self
+    sourceLanguagePopup.action = #selector(sourcePopupSelected(item:))
+    targetLanguagePopup.action = #selector(targetPopupSelected(item:))
+
+    // Restore settings
+    let sourceLanguage = settings.string(forKey: SettingsKey.SourceLanguage) ?? DEFAULT_SOURCE.0
+    let targetLanguage = settings.string(forKey: SettingsKey.TargetLanguage) ?? DEFAULT_TARGET.0
+    sourceLanguagePopup.setTitle(Constants.findSourceLabel(sourceLanguage) ?? DEFAULT_SOURCE.1)
+    targetLanguagePopup.setTitle(Constants.findTargetLabel(targetLanguage) ?? DEFAULT_TARGET.1)
+  }
```

Comment:
When it becomes highly functional, it may be better to separate codes written on AppDelegate class into other model classes or controllers in order to separate duties.

Reply:
参考(In Japanese): https://qiita.com/nori0620/items/66ebc623f63fc3f0ca20

Reply:
Leave this for now but should be fixed in another PR